### PR TITLE
GP2-3856 - Core components JavaScript build

### PR DESCRIPTION
**Pull request by bobmeredith**
First part of migrating all builds to webpack.
the aim is a single build to cover all Magna and logged-out sites, so ideally, we want 4 files - js and css for each site.
Sadly, the domestic and components css builds produce many separate .css files, each named from the entry scss file.
It's a big job to consolidate this, so this PR covers just the components build from core/components and a main .css build from domestic/sass.
I've renamed the distribution files to make them simpler.

**Magna build:**
unchanged apart from outputs being renamed
magna.js, magna_styles.css

**/core/components:**
components.js and components_styles.css

**/domestic/:**
loggedout.js, loggedout_styles.css

with corresponding maps 

the core/components build is actually just the cookie modal.

### Workflow

- [x] Ticket exists in Jira https://uktrade.atlassian.net/browse/GP2-3856
- [x] Jira ticket has the correct status.
- [x] [Changelog](CHANGELOG.md) entry added.

### Reviewing help

- [ ] Explains how to test locally, including how to set up appropriate data
- [ ] Includes screenshot(s) - ideally before and after, but at least after

### Housekeeping

- [ ] Added all new environment variables to Vault.
- [ ] Cleaned up old feature flags
- [ ] Upgraded any vulnerable dependencies.
- [ ] Ran the `make manage download_geolocation_data` command
- [ ] I have updated security dependencies
- [ ] Python requirements have been re-compiled.
- [ ] Frontend assets have been re-compiled.

### Merging

- [x] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)

### CODE REVIEW

File: core/components/gulpfile.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ gulp.task(
 )
 
 // // Export-elements-specific component styling
-
+/* No longer needed
```
Comment (froddd) [inline core/components/gulpfile.js:30]: Get rid?

File: react-components/src/components/CookiesModal/CookiesModal.test.jsx [hunk 2]
```diff
@@ -25,6 +23,9 @@ const defaultProps = {
 describe('CookiesModal', () => {
 
   test('handles not being shown', () => {
+    global.document.cookie = "cookie_preferences_set=true"; //set cookies in whatever format you like
+    document.cookie = "cookie_preferences_set=true";
+    CookiesManager.getPreferencesCookie.mockImplementation(() => true)
```
Comment (froddd) [inline react-components/src/components/CookiesModal/CookiesModal.test.jsx:28]: If `getPreferencesCookies` is mocked, is it necessary to actually place a cookie in the test `document`? Seems like one of these only would signify intent a bit better.

File: react-components/src/components/CookiesModal/dit.components.cookie-notice.js [hunk 3]
```diff
@@ -120,6 +120,7 @@ dit.components.cookieNotice = function() {
   }
 
   function getPreferencesCookie () {
+    console.log('Cookies - in gerPreferencescookie', document.cookie, getCookie(cookiePreferencesName))
```
Comment (froddd) [inline react-components/src/components/CookiesModal/dit.components.cookie-notice.js:123]: Probably remove :)

File: react-components/webpack.config.js [hunk 4]
```diff
@@ -49,9 +53,10 @@ module.exports = {
         exclude: /node_modules/,
       },
       {
-        test: /\.s?css$/i,
+        test: /\.scss$/i,
+        exclude: /node_modules/,
         use: [
-          // extract to seperate file
+          // extract to separate file
```
Comment (froddd) [inline react-components/webpack.config.js:59]: 🙌 😄

**Comment by seijihg** [comment]
there is a licence file added.

**Review by froddd — COMMENTED** [review]
A couple of minor code cleanup tweaks.

Not sure if related, but the cookie preferences page doesn't quite seem to work as intended: when loading after accepting all cookies, none of the preferences are shown. Selecting any combination of preferences seems to always produce (or retain?) the same preferences in the cookie:

```
{"essential":true,"settings":true,"usage":true,"campaigns":true}
```

File: react-components/webpack.config.js (hunk 4 above)
Comment (bobmeredith) [reply react-components/webpack.config.js:59]: You'll always find a rat in separate!

File: react-components/src/components/CookiesModal/CookiesModal.test.jsx (hunk 2 above)
Comment (bobmeredith) [reply react-components/src/components/CookiesModal/CookiesModal.test.jsx:28]: I remember struggling with this - there seemed to be some cross-contamination between tests.
I'll have a look.

**Review by froddd — APPROVED** [review]
Code looks good -- still an issue with cookie preferences.

File: core/components/gulpfile.js (hunk 1 above)
Comment (bobmeredith) [reply core/components/gulpfile.js]: Yeah - I was always intending to kill the whole file

**Comment by bobmeredith** [comment]
@froddd  Yeah, thanks for checking that. There was an issue - I needed to include an extra file in the components bundle.
All better now.

**Pull request merged**